src/loader/cmti.ml

```diff
@@ -529,8 +554,19 @@ and read_module_type_declaration env parent mtd =
 
 and read_module_declaration env parent md =
   let open Module in
-  let name = parenthesise (Ident.name md.md_id) in
-  let id = `Module(parent, Odoc_model.Names.ModuleName.of_string name) in
+  let id =
+#if OCAML_MAJOR = 4 && OCAML_MINOR >= 10
+    match md.md_id with
+    | Some id ->
+      let name = parenthesise (Ident.name id) in
+      `Module(parent, Odoc_model.Names.ModuleName.of_string name)
+    | None ->
+      `Module(parent, Odoc_model.Names.ModuleName.of_string "_")
+#else
```

Comment:
I don't think this is the right behaviour for the no id case. In such cases we should just not produce a module item. That probably means that `module_declaration` should return a list of items like some of the other similar functions (or an `option` but that is probably slightly more effort).

Reply:
Done.